### PR TITLE
WT-9796 Add debugging for logging getting stuck.

### PR DESCRIPTION
**Pull request by sueloverso**
@agorrod and @mickgraham-mongodb here's some debugging for logging. I record the 10 most recent timestamps and LSNs. In gdb you'd be able to see how many times it looped as well.

### CODE REVIEW

**Comment by sueloverso** [comment]
@mickgraham-mongodb I redid the debugging I added with your new information that it is on the sync_lsn. I record the session address of the thread that does the release and I save a copy of the log slot at various times so that we can see if/that the flags are set to figure out why that releasing thread is not waiting in `wt_log_release`.

File: src/log/log.c
Comment (agorrod) [inline src/log/log.c:1864]: Adding a barrier here unnecessarily doesn't feel amazing, but I'd take it if it means we can diagnose logging hangs more readily.

**Review by agorrod — COMMENTED** [review]
Seems like a good idea to me - the equivalent in WT_REF tracking has been immensely valuable.
Don't wait on me for a final review, and sorry for the delay. The ping went into the ether somewhere, I check assigned reviews to look for things waiting on me..

File: src/log/log.c
Comment (sueloverso) [reply src/log/log.c:1864]: It is temporary for debugging. This one here is to determine if that is really the problem. If we determine that the barrier on the slot flags is the issue, then I think the right place to add it would be to the code where we set the sync flag on the slot, not here. We set sync on the slot far less often so we'd want it to be on that uncommon code path rather than here, I agree.

File: src/log/log.c
Comment (sueloverso) [inline src/log/log.c:1983]: We want this change even if we throw away this debug branch. Actually, we should consider a `wt_log_release` refactor to call `wt_log_force_sync` to have one place where logging syncs.

File: src/log/log.c
Comment (sueloverso) [reply src/log/log.c:1864]: I am going to put all of this under HAVE_DIAGNOSTIC so that this debugging code/barrier does not get into any MongoDB production code.

**Comment by sueloverso** [comment]
@mickgraham-mongodb I added you back on as a reviewer. Alex is out for a while, I cannot review a PR I created and we cannot (or should not) merge without one approved reviewer. You can review/approve even though you contributed to the branch.

File: src/log/log.c
Comment (mickgraham-mongodb) [reply src/log/log.c:1985]: Raised WT-10206 to do the refactor. With your HAVE_DIAGNOSTIC change we are good to have this in.

**Review by mickgraham-mongodb — APPROVED** [review]

**Pull request merged**
